gdsfactory/routing/auto_taper.py

```diff
@@ -86,4 +86,6 @@ def auto_taper_to_cross_section(
         )
     taper_ref = component.add_ref(taper_component)
     taper_ref.connect(p0.name, port)
-    return taper_ref.ports[p1.name]
+    port_new = taper_ref.ports[p1.name]
+    port_new.name = port.name
```

Comment:
This is stricly speaking save because ref is returning a copy of a port, but I would suggest to still go the good practice way of 
```suggestion
    port_new = taper_ref.ports[p1.name].copy()
    port_new.name = port.name
```